src/libraries/System.Memory/ref/System.Memory.cs

```diff
@@ -430,6 +434,12 @@ public static partial class MemoryExtensions
             public bool AppendFormatted(string? value) { throw null; }
             public bool AppendFormatted(string? value, int alignment = 0, string? format = null) { throw null; }
         }
+        public ref struct SpanSplitEnumerator<T> where T : System.IEquatable<T>
```

Comment:
I am just thinking, would it be sensible to move this into the System namespace directly? If someone doesn't like var, they will have to write.
`MemoryExtensions.SpanSplitEnumerator<char> enumerator = span.Split(' ');`
The original implementation (that got reverted) had the enumerator in System directly, I changed it as that was what was approved in api review.

Reply:
Yeah I suppose it wouldn't matter for the `foreach` use-case 👍

Reply:
Do you see any circumstances where it may be passed into other methods? To then `foreach` over it (or otherwise) in there? As that would result in you having to declare the type as part of the parameter list. Though will probably also be relatively uncommon. Just trying to avoid any possible DX issues with the api

Reply:
@terrajobst, @bartonjs, opinions on the nested enumerator here? It looks like for MemoryExtensions.EnumerateRunes and MemoryExtensions.EnumerateLines we put their return enumerator types (SpanRuneEnumerator and SpanLineEnumerator) as top-level types in System.Text.

Reply:
The guidance on nested is more or less "don't do it, unless almost no one will ever see the type name", and are based on the CLR languages as a whole, not how convenient they are in just C#. 

It can't be used with System.Linq.Enumerable, because it can't be boxed; so it's really a question of "would a reasonable someone want to interact with this manually?"  If yes, then top level type; if "no, just the unit tests" then nested is OK.

I'm not sure if we had actual scenarios for the other two; or if we were just going with "top level is better experience for all languages".  My fail-safe here would be to say top-level if we're unsure.

Reply:
> My fail-safe here would be to say top-level if we're unsure.

Ok. I'm moving SpanSplitEnumerator to be next to SpanLineEnumerator and SpanRuneEnumerator in System.Text.

Reply:
I hope I dont sound too contrarian, but is System.Text the right namespace for this? If it was a non-generic `char` only version it would seem fine. But I feel like it is a little odd importing the `System.Text` namespace in order to split on numeric values for instance. Or some custom reference type. For `SpanRuneEnumerator` it makes sense as it is non-generic and `Rune` is inherently a text primitive.

Reply:
System is too broad. Where would you expect it? If there's no better option I'll just make it nested again.

Reply:
Not sure where I'd expect it to be honest. Only other place I could imagine is `System.Buffers` potentially.

Reply:
`System.Buffers` wouldnt work?

```
Contains types used in creating and managing memory buffers, such as those represented by Span<T> and Memory<T>.
```

This is the description of the namespace on msdn. Sounds somewhat appropriate,  it technically manages a memory buffer.

Reply:
That's a stretch in my opinion. And having the type be nested works just fine and it's what we all previously agreed to.

Reply:
I see, makes sense, thanks!